### PR TITLE
Update qchem functions to use Molecule 

### PR DESCRIPTION
**Pull request by soranjh**
The `molecular_hamiltonian` function is updated to accept a molecule object to match recent upgrades in qchem. All molecular coordinates are updated to have a `(N * 3)` where `N` is the number of atoms.

### CODE REVIEW

**Comment by soranjh** [comment]
[sc-59836]

**Review by ddhawan11 — COMMENTED** [review]
Looks good @soranjh.

File: demonstrations/tutorial_mol_geo_opt.py [hunk 1]
```diff
@@ -126,7 +126,8 @@
 
 
 def H(x):
-    return qml.qchem.molecular_hamiltonian(symbols, x, charge=1)[0]
+    molecule = qml.qchem.Molecule(symbols, x, charge=1)
+    return qml.qchem.molecular_hamiltonian(molecule)[0]
```
Comment (obliviateandsurrender) [inline demonstrations/tutorial_mol_geo_opt.py:130]: General comment - This is a nice example of where the older UI looked cleaner.

**Review by obliviateandsurrender — APPROVED** [review]
Thanks @soranjh, looks good to me!

**Review by ddhawan11 — APPROVED** [review]
Looks good @soranjh

**Pull request merged**